### ISSUE TITLE
GCC support in CModule

### ISSUE THREAD

**Issue by mephi42** [frida/frida-gum#510]
If one wants to use CModule on architectures that aren't supported by TCC, one option is to use the GNU toolchain (gcc+ld+objcopy+nm) for code generation and inspection instead. This may also be useful in cases where good optimization is desired.

I have a working prototype at https://github.com/mephi42/frida-gum/tree/gcc, where just a couple tests still fail.
Would it be interesting to have this upstream?

**Comment by oleavr** [comment]
Yeah I have actually been thinking about this occasionally so am super-excited to see your prototype. 🎉 Would love to support this upstream!
